genai-perf/tests/test_synthetic_data_retriever.py

```diff
@@ -139,3 +139,41 @@ def test_synthetic_multiple_files(self, mock_prompt, mock_image):
                 assert len(row.images) == 1
                 assert row.texts[0] == "test prompt"
                 assert row.images[0] == "data:image/jpeg;base64,test_base64_encoding"
+
+    @patch(
+        "genai_perf.inputs.retrievers.synthetic_data_retriever.SyntheticPromptGenerator.create_synthetic_prompt",
+        return_value="test prompt",
+    )
+    @patch(
+        "genai_perf.inputs.retrievers.synthetic_data_retriever.SyntheticPromptGenerator.create_system_prompts_pool"
+    )
+    @patch(
+        "genai_perf.inputs.retrievers.synthetic_data_retriever.SyntheticPromptGenerator.get_random_system_prompt",
+        return_value="system prompt",
+    )
+    def test_synthetic_with_system_prompts(
```

Comment:
Can we have expected output somewhere in the test and compare that with actual data? I think it's a bit hard to understand what the test is doing and what it's trying to check.

Reply:
I generally would actually rather move away from that approach. It's quite brittle and hard-to-maintain to provided expected outputs and check them exactly against the returned outputs. I'm trying to find tests where we do that and see what the alternatives are.

If that's really important to you and @debermudez, I can add expected output as code or a comment, but it feels redundant to me and adds more required maintenance. I think the name of the test specifies its purpose and the asserts were meant to be self-commenting.

Reply:
I said comparing an actual vs expected, but what I really wanted to say is readability. And comparing actual vs expected is just one easy way to achieve that. A lot of tests for converters do that as well for different cases. Some of the parts that I think was hard while reading this test were:
1. Magic numbers (e.g. hard coded `[0]` indexing)
2. Checking the contents of `dataset` part (this part assumes the reader has the schema of `dataset` in their head)
```python
# dataset --> access 'files_data` --> index with `synthetic_input_filenames[0]` --> access 'rows'
for row in dataset.files_data[synthetic_input_filenames[0]].rows:
    assert row.texts[0].startswith("system prompt ")
```

Reply:
That's fair. I'll rework this test to be clearer about what we're checking for. Let me spend some time thinking about a way to do so without tying it too closely with the current schema.

Thanks for pointing this out, Hyunjae.